modules/basicfuncs/list-funcs.c

```diff
@@ -151,6 +164,8 @@ _list_nth(gint argc, GString *argv[], GString *result, gint ndx)
   list_scanner_init(&scanner);
   list_scanner_input_gstring_array(&scanner, argc, argv);
 
+  _adjust_list_index(argc, argv, &ndx);
+
   i = 0;
   while (i < ndx && list_scanner_scan_next(&scanner))
```

Comment:
I don't know what is the desired behaviour in case of index errors, but with the original code in case of "index overflow" we will do nothing.
After this loop `i == ndx`, and the `list_scanner_scan_next` will return FALSE in the next if statement.

With this modification in case of negative "overflow". i.e: `length == 3; ndx == -4`
`_adjust_list_index` will produce: `ndx == -1`
In the loop we will exit with `i==0 > ndx==-1`.
And in the next if statement: `i==0 > ndx == -1` && `list_scanner_scan_next` will return TRUE and set the position to the first element.

Reply:
This was a good catch, and indeed my testcases were not enough to catch this. I have now added a couple more tests and also fixed the code.

Thanks for spotting this.